Index: activeweb-testing/src/test/java/org/javalite/activeweb/InjectionFilterJUnit5Spec.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- activeweb-testing/src/test/java/org/javalite/activeweb/InjectionFilterJUnit5Spec.java	(date 1549385688000)
+++ activeweb-testing/src/test/java/org/javalite/activeweb/InjectionFilterJUnit5Spec.java	(date 1549385688000)
@@ -0,0 +1,55 @@
+/*
+Copyright 2009-2016 Igor Polevoy
+
+Licensed under the Apache License, Version 2.0 (the "License"); 
+you may not use this file except in compliance with the License. 
+You may obtain a copy of the License at 
+
+http://www.apache.org/licenses/LICENSE-2.0 
+
+Unless required by applicable law or agreed to in writing, software 
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+
+package org.javalite.activeweb;
+
+
+import app.services.RedirectorModule;
+import com.google.inject.Guice;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Igor Polevoy
+ */
+class InjectionFilterJUnit5Spec extends AppIntegrationSpec {
+
+
+    InjectionFilterJUnit5Spec(){
+        suppressDb();
+    }
+
+    @BeforeEach
+    void before(){
+        setInjector(Guice.createInjector(new RedirectorModule()));
+        setTemplateLocation("src/test/views");
+    }
+    
+    @Test
+    void shouldNotRedirectFromFilterBecauseNoRedirectTargetSupplied(){
+        controller("abc_person").get("index");
+        a(redirectValue()).shouldBeNull();
+    }
+
+    @Test
+    void shouldRedirectFromFilterBecauseRedirectTargetSupplied(){
+        controller("abc_person").param("target", "Google").get("index");
+        a(redirectValue()).shouldBeEqual("http://google.com");
+
+        controller("abc_person").param("target", "Yahoo").get("index");
+        a(redirectValue()).shouldBeEqual("http://yahoo.com");
+    }
+}
